src/main/java/org/javarosa/core/services/storage/IStorageUtilityIndexed.java

```diff
@@ -207,6 +207,19 @@
      */
     E getRecordForValue(String metaFieldName, Object value) throws NoSuchElementException, InvalidIndexException;
 
+
+    /**
+     * Retrieves a Vector of Externalizable objects from the storage for which, for each field in fieldNames,
+     * the record has the correct corresponding value in values
+     *
+     * @param metaFieldNames A list of metadata field names to match
+     * @param values     The values which must match the field names provided
+     *
+     * @return A Vector of Externalizable objects e, such that the fields specified are equal to the corresponding values provided.
+     *
+     */
+    Vector<E> getRecordsForValues(String[] metaFieldNames, Object[] values);
```

Comment:
Is this the same signature as the custom implementations in Android AND formplayer? Is good to finally have it committed in code.